main.go

```diff
@@ -2804,6 +2810,61 @@ The following options are recognized:
 		return nil
 	})
 
+	r.Dispatch("uuid", &Help{
+		Summary: "Generate a version 4 uuid value and write it to the specified path and/or key. If no key is provide the key `uuid` is used",
+		Usage:   "safe uuid PATH[:KEY]",
+		Type:    NonDestructiveCommand,
+		Description: ``,
+	}, func(command string, args ...string) error {
+		rc.Apply(opt.UseTarget)
+
+		if len(args) == 0 {
+			r.ExitWithUsage("uuid")
+		}
+
+		u := uuid.NewRandom()
+
+		stringuuid := u.String()
+
+		v := connect(true)
+
+		for len(args) > 0 {
+			var path, key string
+			if vault.PathHasKey(args[0]) {
+				path, key, _ = vault.ParsePath(args[0])
+				args = args[1:]
+			} else {
+				path, key = args[0], "uuid"
+				//If the key looks like a full path with a :key at the end, then the user
+				//probably botched the args
+				if vault.PathHasKey(key) {
+					return fmt.Errorf("For secret `%s` and key `%s`: key cannot contain a key", path, key)
+				}
+
+			}
+			s, err := v.Read(path)
+			if err != nil && !vault.IsNotFound(err) {
+				return err
+			}
+			exists := (err == nil)
+			if opt.SkipIfExists && exists && s.Has(key) {
+				if !opt.Quiet {
+					fmt.Fprintf(os.Stderr, "@R{Cowardly refusing to update} @C{%s:%s} @R{as it is already present in Vault}\n", path, key)
+				}
+				continue
```

Comment:
if the loop goes away, this becomes a return

Reply:
set return statement to return err.

Reply:
*continue statement, but yes